sherpa/bin/streaming_pruned_transducer_statelessX/streaming_server.py

```diff
@@ -483,8 +507,19 @@ async def handle_connection_impl(
 
             while len(stream.features) > self.chunk_length:
                 await self.compute_and_decode(stream)
-                result = self.beam_search.get_texts(stream)
-                await socket.send(result)
+                hyp = self.beam_search.get_texts(stream)
+
+                if stream.endpoint_detected(self.online_endpoint_config):
+                    final = 1
+
+                message = {
+                    "text": hyp,
+                    "final": final,
+                }
+
+                await socket.send(json.dumps(message))
+                if final:
```

Comment:
Maybe we shouldn't return here. I thought the goal of an endpointer is to provide additional information to show sth like VAD segments/pauses in continuous speech? If we return here, the connection would close on the first segment.

Reply:
This makes sense I think.  the way it was done in Kaldi was for demonstration purposes, really.

Reply:
I see. 
In terms of the endpointer functionality, after slightly modifying it to not return on the first segment and resetting `stream.hyp` as follows:
![snap 2022-09-07 at 11 46 27 PM](https://user-images.githubusercontent.com/31003116/189053491-25c960da-d51a-4cf3-9e38-84a95b135be4.png)

the endpointing detection works fine in the beginning but seems to start showing unexpected behaviour at some point, as you can see towards the bottom of the screenshot, many endpoints were detected:

![snap 2022-09-07 at 11 43 45 PM](https://user-images.githubusercontent.com/31003116/189053291-9317001c-84fe-4eae-89d0-06a4aceea3fc.png)

Maybe there's a bug somewhere in the endpointer?

Reply:
There are three default rules in endpoint detection. If any of them is activated, then we assume endpoint is detected.

I suspect that rule 3 is triggered.

You can find the 3 rules in
https://github.com/k2-fsa/sherpa/blob/9e5bc4ec7ff6fffb95b932cb092b4209932ad20f/sherpa/python/sherpa/online_endpoint.py#L29-L49

---

We provide the following commandline arguments to change them:
https://github.com/k2-fsa/sherpa/blob/9e5bc4ec7ff6fffb95b932cb092b4209932ad20f/sherpa/python/test/test_online_endpoint.py#L116-L118

---

You can pass `--help` to see the usage.

Reply:
I see, thanks! I have no further questions.